extended/build.gradle

```diff
@@ -43,8 +43,15 @@ shadowJar {
     archiveClassifier.set('extended')
 }
 
+// to init the submodule in case of new submodule or refresh it if exists, every time we compile 
+task gitSubmoduleLoad {
+    def execute = ["bash", "-c", "git submodule status apoc-core | grep -q \"^-\" && rm -r apoc-core && git submodule update --init apoc-core || git submodule update --remote apoc-core"]
+            .execute()
+    execute.waitForProcessOutput(System.out, System.err)
```

Comment:
Two questions on this:

1. What does the `grep -q \"^-\"` part do?
2. Why do we need `rm -r apoc-core && git submodule update --init apoc-core` both here and in the init script?

Reply:
@vga91 Nacho couldn't explain this to me, can you?

Reply:
@Lojjs 

1. The `git submodule status apoc-core`, initially it has an output like this:
```
-COMMIT_HASH apoc-core
```
where the first `-` sign indicates that the submodule is not initialized.
Therefore, with `grep -q \"^-\"`, I'll check whether the module is initialized (and in this case I'll do `rm -r apoc-core && git submodule update --init apoc-core`) or not (then `git submodule update --remote apoc-core` will be executed).
It's the simplest method I've found to automatically check if the module is initialized and execute the correct command.

2. Because I could execute `Load gradle changes` (e.g. via IntelliJ), and in this case this piece of code will be executed, and the init script would be useless.

And therefore, every time I want to update the core changes, it would be enough to do "Load gradle changes"

Otherwise, with the CI (or by just doing `git clone <repo>` and `./gradlew shadow`), this piece of code would not be executed, so we need to manually run` init.gradle` (via `./gradlew  --init-script init.gradle build/shadow/otherCommand`)

Reply:
Thanks for the explanation, now I understand :)